server/archive.api.js

```diff
@@ -0,0 +1,63 @@
+const HomebrewModel = require('./homebrew.model.js').model;
+const router = require('express').Router();
+const asyncHandler = require('express-async-handler');
+
+const archive = {
+	archiveApi : router,
+	/* Searches for matching title, also attempts to partial match */
+	findBrews  : async (req, res, next)=>{
+		try {
+			console.table(req.query);
+
+			const title = req.query.title || '';
+			const page = Math.max(parseInt(req.query.page) || 1, 1);
+			const minPageSize = 6; 
+			const pageSize = Math.max(parseInt(req.query.size) || 10, minPageSize);  
+			const skip = (page - 1) * pageSize;
+			
+			const titleQuery = {
+				title     : { $regex: decodeURIComponent(title), $options: 'i' },
+				$or: [],
+				published : true
+			};
+
+			if (req.query.legacy === 'true') {
+                titleQuery.$or.push({ renderer: 'legacy' });
+            };
+
+            if (req.query.v3 === 'true') {
+                titleQuery.$or.push({ renderer: 'V3' });
+            };
+
+			const projection = {
+				editId   : 0,
+				googleId : 0,
+				text     : 0,
+				textBin  : 0,
+			};
+			const brews = await HomebrewModel.find(titleQuery, projection)
+                .skip(skip)
+                .limit(pageSize)
+                .maxTimeMS(5000)
+                .exec();
+
+			const totalBrews = await HomebrewModel.countDocuments(titleQuery, projection).maxTimeMS(5000);
```

Comment:
Do we need to pass in `projection` here, as we want to count all documents matching `titleQuery` and all those documents will have a `title` and the `published` flag is part of `titleQuery`. That is, there is nothing in `projection` that says "don't count documents that don't have these fields".

Reply:
you're absolutely right